test/e2e/healthcheck_hybrid_test.go

```diff
@@ -0,0 +1,155 @@
+		Expect(err).ToNot(HaveOccurred())
+		ingressToken, err := global.GetKuma().GenerateDpToken("default", "ingress")
+		Expect(err).ToNot(HaveOccurred())
+
+		optsRemoteUniversal = []DeployOptionsFunc{
+			WithGlobalAddress(global.GetKuma().GetKDSServerAddress()),
+		}
+		remoteUniversal = universalClusters.GetCluster(Kuma3)
+		err = NewClusterSetup().
+			Install(Kuma(core.Remote, optsRemoteUniversal...)).
+			Install(EchoServerUniversal("dp-echo-1", "default", "echo-universal-1", echoServerToken, WithProtocol("tcp"))).
+			Install(EchoServerUniversal("dp-echo-2", "default", "echo-universal-2", echoServerToken, WithProtocol("tcp"), ProxyOnly(), ServiceProbe())).
+			Install(EchoServerUniversal("dp-echo-3", "default", "echo-universal-3", echoServerToken, WithProtocol("tcp"))).
+			Install(IngressUniversal("default", ingressToken)).
+			Setup(remoteUniversal)
+		Expect(err).ToNot(HaveOccurred())
+		err = remoteUniversal.VerifyKuma()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		if ShouldSkipCleanup() {
+			return
+		}
+
+		_ = k8s.KubectlDeleteFromStringE(remoteK8s.GetTesting(), remoteK8s.GetKubectlOptions(), namespaceWithSidecarInjection(TestNamespace))
+		err := remoteK8s.DeleteKuma(optsRemoteK8s...)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = remoteUniversal.DeleteKuma(optsRemoteUniversal...)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = global.DeleteKuma()
+		Expect(err).ToNot(HaveOccurred())
+
+		err = remoteUniversal.DismissCluster()
+		Expect(err).ToNot(HaveOccurred())
+	})
+
+	It("should not load balance requests to unhealthy instance", func() {
+		pods, err := k8s.ListPodsE(remoteK8s.GetTesting(), remoteK8s.GetKubectlOptions(TestNamespace), metav1.ListOptions{
+			LabelSelector: fmt.Sprintf("app=%s", "demo-client"),
+		})
+		Expect(err).ToNot(HaveOccurred())
+		Expect(pods).To(HaveLen(1))
+
+		var counter1, counter2, counter3 int
+		for i := 0; i < 100; i++ {
+			var stdout string
+			cmd := []string{"curl", "-v", "-m", "3", "--fail", "echo-server_kuma-test_svc_8080.mesh"}
+			_, err := retry.DoWithRetryE(
+				remoteK8s.GetTesting(),
+				fmt.Sprintf("kubectl exec -- %s", strings.Join(cmd, " ")),
+				DefaultRetries,
+				DefaultTimeout,
+				func() (string, error) {
+					var err error
+					stdout, _, err = remoteK8s.Exec(TestNamespace, pods[0].GetName(), "demo-client", cmd...)
+					if err != nil {
+						Expect(err).ToNot(MatchError("command terminated with exit code 56"))
+						return "", nil
+					}
+					return "", err
+				},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			switch {
+			case strings.Contains(stdout, "Echo echo-universal-1"):
+				counter1++
+			case strings.Contains(stdout, "Echo echo-universal-2"):
+				counter2++
+			case strings.Contains(stdout, "Echo echo-universal-3"):
+				counter3++
+			}
+		}
+		Expect(counter2).To(Equal(0))
+		Expect(math.Abs(float64(counter3-counter1)) <= 4).To(BeTrue())
```

Comment:
what does it mean? why counter3 - counter1 <= 4?

Reply:
I probably should rewrite this using percentage, I was about to check that request aren't always evenly distributed, it could be 52 and 48, but most of the time it's 50/50

Reply:
Yeah standard deviation and such.